klsh/hamming_ann.py

```diff
@@ -156,12 +158,14 @@ def query(self, X, k, return_dist=False):
             dist = np.zeros_like(nbrs)
 
         M, N = self._P_compact_sorted.shape
-
+        
+        X_compact=packbits_axis(X_compact[:,self._P_indices])
+        
         # TODO: MAKE THIS MORE EFFICIENT
         for i, val in enumerate(X_compact):
             # find ordered index within each random permutation
             P_indices = np.array([np.searchsorted(self._P_compact_sorted[j],
-                                                  val) for j in range(M)])
+                                                  val[j]) for j in range(M)])
```

Comment:
Why the change from ``val`` to ``val[j]``? Previously all the values were searched; now only a single value in each row is searched.